core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/AwsS3V4HttpProperties.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal.signer;
+
+import static software.amazon.awssdk.http.auth.aws.AwsV4HttpSigner.CHUNKED_ENCODING;
+import static software.amazon.awssdk.http.auth.aws.AwsV4HttpSigner.PAYLOAD_SIGNING;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.STREAMING_UNSIGNED_PAYLOAD_TRAILER;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.safeParseLong;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.validatedProperty;
+
+import java.time.Clock;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.Header;
+import software.amazon.awssdk.http.auth.aws.AwsS3V4HttpSigner;
+import software.amazon.awssdk.http.auth.aws.checksum.ChecksumAlgorithm;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4HttpProperties;
+import software.amazon.awssdk.http.auth.aws.util.CredentialScope;
+import software.amazon.awssdk.http.auth.aws.util.HttpChecksumUtils;
+import software.amazon.awssdk.http.auth.spi.SignRequest;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+
+/**
+ * An extension of {@link AwsV4HttpProperties}, which provides access to more specific parameters
+ * used by {@link AwsS3V4HttpSigner}.
+ */
+@SdkProtectedApi
+interface AwsS3V4HttpProperties extends AwsV4HttpProperties {
+
+    static AwsS3V4HttpProperties create(SignRequest<?, ? extends AwsCredentialsIdentity> signRequest) {
+        boolean payloadSigning = validatedProperty(signRequest, PAYLOAD_SIGNING, false);
```

Comment:
The parent interface (`AwsV4Properties`) doesn't have payload-signing or chunked-encoding properties. But, yeah, it's fine if the resolver sets the properties, the default here just reflects the current behavior (in `AwsS3V4SignerParams`).

Reply:
The AwsV4HttpSigner interface says that payload signing default is true. So without setting the property if one gets the behavior of payload signing = false, that doesn't seem right. 

The interface doesn't communicate that payload signing will be false, with chunked encoding is true. We could make that callout in the interface docs, but then that's putting an S3-ism in the generic interface, which I don't think is correct - it would steer me towards an S3 specific signer in S3's module then, maybe.

Reply:
For the s3 signer, if payload signing isn't set and chunk-encoding is true, it still doesn't do chunk-encoding, because it doesn't have a concept of unsigned chunk-encoding.

Reply:
> it doesn't have a concept of unsigned chunk-encoding

If our interface doesn't support this (or any) combination, the sign/signAsync methods should throw when the combination is what the user asks for, v/s silently not do on of the things they asked for.